### PR TITLE
Divide ph 1

### LINKED ISSUE #15: Split extraction phase 1 code
**Is your feature request related to a problem? Please describe.**
The steps to generate a) txl-based function and struct identification and b) codequery based function-call-graph and map-graph should be split into 2 steps. This is because step-1 is for entire repo and can be reused for all functions. Whereas step-2 is extraction target specific.

**Describe the solution you'd like**
Split extraction_runner.py to 2 separate executors and make the 2nd one take first's artefacts as input if necessary.

**Describe alternatives you've considered**
None.

**Additional context**
None.

### PR DESCRIPTION
**Pull request by sdsen**
fixes #15.

### CODE REVIEW

**Comment by palanik1** [comment]
@sdsen can you pull in the latest from the main branch and raise the PR

**Comment by sdsen** [comment]
done

File: extraction_runner.py [hunk 1]
```diff
@@ -22,7 +23,7 @@ def check_if_cmd_available():
     return True
 
 def check_if_file_available():
-    files = [r'asset/c-extract-functions.txl', r'asset/c-extract-struct.txl', r'asset/c.grm.1', r'asset/bom.grm', r'asset/helper_hookpoint_map.json']
+    files = []
```
Comment (palanik1) [inline extraction_runner.py:26]: may be make files an input parameter?

File: foundation_maker.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
+# Author Info:
+# Sayandeep Sen (sayandes@in.ibm.com)
+# Palani Kodeswaran (palani.kodeswaran@in.ibm.com)
+
```
Comment (palanik1) [inline foundation_maker.py:4]: can you rename file to something like annotator.py or something  more relevant..

**Review by palanik1 — CHANGES_REQUESTED** [review]

File: extraction_runner.py (hunk 1 above)
Comment (sdsen) [reply extraction_runner.py:26]: these are internal files ... in the sense, that user of the extractor doesn't need to be aware of them. So don't want to specify them as input. If you want to not have them hardcoded, we can move them to a config.json (in a future fix)?

File: foundation_maker.py (hunk 2 above)
Comment (sdsen) [reply foundation_maker.py]: fixed.

File: extraction_runner.py (hunk 1 above)
Comment (palanik1) [reply extraction_runner.py:26]: files is being set to empty and then iterated below no?

Comment (sdsen) [reply extraction_runner.py:26]: Yes .. all internal  files  based processing now moved to annotater.py . I can remove the function as well and resurface once we need internal files for processing. 

Shall I do that?

Comment (palanik1) [reply extraction_runner.py:26]: Yep..

Comment (sdsen) [reply extraction_runner.py]: fixed.

**Pull request merged**